src/manifolds/ShapeSpace.jl

```diff
@@ -0,0 +1,241 @@
+    check_point(M::KendallsPreShapeSpace, p; atol=sqrt(max_eps(X, Y)), kwargs...)
+
+Check whether `p` is a valid point on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_point(M::KendallsPreShapeSpace, p; atol=sqrt(eps(eltype(p))), kwargs...)
+    for p_row in eachrow(p)
+        if !isapprox(mean(p_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(p_row),
+                "The point $(p) does not lie on the $(M) since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+"""
+    check_vector(M::KendallsPreShapeSpace, p, X; kwargs... )
+
+Check whether `X` is a valid tangent vector on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_vector(M::KendallsPreShapeSpace, p, X; atol=sqrt(max_eps(X, Y)), kwargs...)
+    for X_row in eachrow(X)
+        if !isapprox(mean(X_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(X_row),
+                "The vector $(X) is not a tangent vector to $(p) on $(M), since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+embed(::KendallsPreShapeSpace, p) = p
+embed(::KendallsPreShapeSpace, p, X) = X
+
+function get_embedding(::KendallsPreShapeSpace{N,K}) where {N,K}
+    return ArraySphere(N, K)
+end
+
+@doc raw"""
+    manifold_dimension(M::KendallsPreShapeSpace)
+
+Return the dimension of the [`KendallsPreShapeSpace`](@ref) manifold `M`. The dimension is
+given by ``n(k - 1) - 1``.
+"""
+manifold_dimension(::KendallsPreShapeSpace{n,k}) where {n,k} = n * (k - 1) - 1
+
+function project!(::KendallsPreShapeSpace, q, p)
+    q .= p .- mean(p, dims=2)
+    q ./= norm(q)
+    return q
+end
+
+function project!(::KendallsPreShapeSpace, Y, p, X)
+    Y .= X .- mean(X, dims=2)
+    Y .-= dot(p, Y) .* p
+    return Y
+end
+
+function Random.rand!(M::KendallsPreShapeSpace, pX; vector_at=nothing, σ=one(eltype(pX)))
+    if vector_at === nothing
+        project!(M, pX, randn(representation_size(M)))
+    else
+        n = σ * randn(size(pX)) # Gaussian in embedding
+        project!(M, pX, vector_at, n) # project to TpM (keeps Gaussianness)
+    end
+    return pX
+end
+function Random.rand!(
+    rng::AbstractRNG,
+    M::KendallsPreShapeSpace,
+    pX;
+    vector_at=nothing,
+    σ=one(eltype(pX)),
+)
+    if vector_at === nothing
+        project!(M, pX, randn(rng, representation_size(M)))
+    else
+        n = σ * randn(rng, size(pX)) # Gaussian in embedding
+        project!(M, pX, vector_at, n) #project to TpM (keeps Gaussianness)
+    end
+    return pX
+end
+
+###
+
+@doc raw"""
+    KendallsShapeSpace{n,k} <: AbstractDecoratorManifold{ℝ}
+
+Kendall's shape space, defined as quotient of a [`KendallsPreShapeSpace`](@ref)
```

Comment:
We could again mention traits, and maybe even mention `QuotientManifold` - to make the connection / helping functions more prominent?

Reply:
I've mentioned the `IsQuotientManifold` trait in the docstring.